volatility/framework/plugins/windows/pslist.py

```diff
@@ -45,21 +46,24 @@ def get_requirements(cls):
         ]
 
     @classmethod
-    def process_dump(cls, context: interfaces.context.ContextInterface, kernel_table_name: str, pe_table_name: str,
-                     proc: interfaces.objects.ObjectInterface) -> interfaces.plugins.FileInterface:
-        """Extracts the complete data for a process as a FileInterface
+    def process_dump(
+            cls, context: interfaces.context.ContextInterface, kernel_table_name: str, pe_table_name: str,
+            proc: interfaces.objects.ObjectInterface,
+            open_method: Type[interfaces.plugins.FileHandlerInterface]) -> interfaces.plugins.FileHandlerInterface:
+        """Extracts the complete data for a process as a FileHandlerInterface
 
         Args:
             context: the context to operate upon
             kernel_table_name: the name for the symbol table containing the kernel's symbols
             pe_table_name: the name for the symbol table containing the PE format symbols
             proc: the process object whose memory should be output
+            open_method: class to provide context manager for opening the file
```

Comment:
Is it important that `open_method` be passed into this function? It seems like it will always be `self.open` from the same class. Is this for cases where another plugin's `open` should be called when calling this plugin's process_dump?

Reply:
Yep, this is a classmethod so it can be called from other plugins.  As such we can't refer to self or instance variables (such as the file handler), so the `open_method` needs to be passed in externally.

Reply:
thanks.